webapp/src/components/chat/ChatResourceList.tsx

```diff
@@ -56,24 +79,117 @@ interface TableItem {
         label: string;
         timestamp: number;
     };
+    tokens: number;
 }
 
-export const ChatResourceList: React.FC<ChatResourceListProps> = ({ chatId }) => {
+export const ChatResourceList: React.FC = () => {
     const classes = useClasses();
     const chat = useChat();
+    const dispatch = useDispatch();
+    const { conversations, selectedId } = useAppSelector((state: RootState) => state.conversations);
+    const { importingDocuments } = conversations[selectedId];
+
     const [resources, setResources] = React.useState<ChatMemorySource[]>([]);
+    const [documentImporting, setDocumentImporting] = React.useState(false);
+    const documentFileRef = useRef<HTMLInputElement | null>(null);
 
     React.useEffect(() => {
-        void chat.getChatMemorySources(chatId).then((sources) => {
-            setResources(sources);
+        const importingResources = importingDocuments
+            ? importingDocuments.map((document, index) => {
+                return {
+                    id: `in-progress-${index}`,
+                    chatId: selectedId,
+                    sourceType: "N/A",
+                    name: document,
+                    sharedBy: "N/A",
+                    createdOn: 0,
+                    tokens: 0,
+                } as ChatMemorySource;
+            })
+            : [];
+
+        setDocumentImporting(importingResources.length > 0);
+        setResources(importingResources);
+
+        void chat.getChatMemorySources(selectedId).then((sources) => {
+            setResources([...importingResources, ...sources]);
         });
+
         // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
looks like we can remove this?

Reply:
We still need this as we don't want to depend on `chat`